src/Nethermind/Nethermind.Consensus.AuRa/AuRaBlockProcessor.cs

```diff
@@ -48,18 +51,38 @@ public class AuRaBlockProcessor : BlockProcessor
             ITxPool txPool,
             IReceiptStorage receiptStorage,
             ILogManager logManager,
-            IAuRaBlockProcessorExtension auRaBlockProcessorExtension)
+            IAuRaBlockProcessorExtension auRaBlockProcessorExtension,
+            ITxPermissionFilter txFilter = null)
             : base(specProvider, blockValidator, rewardCalculator, transactionProcessor, stateDb, codeDb, stateProvider, storageProvider, txPool, receiptStorage, logManager)
         {
             _auRaBlockProcessorExtension = auRaBlockProcessorExtension ?? throw new ArgumentNullException(nameof(auRaBlockProcessorExtension));
+            _logger = logManager?.GetClassLogger<AuRaBlockProcessor>() ?? throw new ArgumentNullException(nameof(logManager));
+            _txFilter = txFilter;
         }
 
         protected override TxReceipt[] ProcessBlock(Block block, IBlockTracer blockTracer, ProcessingOptions options)
         {
+            ValidateTxs(block);
             _auRaBlockProcessorExtension.PreProcess(block, options);
             var receipts = base.ProcessBlock(block, blockTracer, options);
             _auRaBlockProcessorExtension.PostProcess(block, receipts, options);
             return receipts;
         }
+
+        private void ValidateTxs(Block block)
+        {
+            if (_txFilter != null)
```

Comment:
if you set NullFilter as default then you do not have to worry about null checks